pandas/core/generic.py

```diff
@@ -3065,6 +3065,7 @@ def to_csv(
         decimal: Optional[str] = ".",
         errors: str = "strict",
         storage_options: StorageOptions = None,
+        **kwargs,
```

Comment:
I know that @simonjayhawkins and I have a different point of view here but I am strongly against adding kwargs to read_csv. The signature is already massive and adding this only makes things worse

I think we either stick to adding just one keyword arg or just leave this for now

Reply:
The other reason i'm not keen on adding the alias to the signature is that we also have doublequote, escapechar, quotechar and skipinitialspace as parameters to read_csv. So if we have both line_terminator and lineterminator, some bright spark will want snake case equivalents of the others.

Reply:
Would you guys be happy with @jbrockmendel  solution? I think it doesn't add kwargs and `lineterminator` won't appear in the signature

```
def mywrapper(func):
    @wraps(func)
    def new_func(*args, **kwargs):
        if "lineterminator" in kwargs:
            kwargs["line_terminator"] = kwargs.pop("lineterminator")
        return func(*args, **kwargs)
    return new_func

@mywrapper
def read_csv(...)
```

Reply:
A decorator would be a good solution, however some of our decorators 'lose' the signature in the docs e.g. https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_stata.html

Reply:
If the goal is to keep both ad infinitum, it may be possible to mostly use `deprecate_kwarg` and just eat the warning.

Reply:
@simonjayhawkins The "solution" to that issue is probably for the wrapper to rewrite the docstring with an explicit signature that can be created using `Signature` within the docstring.  For example, after post processing the docstring for
```
def foo(x, y):
    """
    Does foo

    Parameters
    ---------
    x : DataFrame
    y : DataFrame
"""
```

would become
```
    """
    foo(x, y)

    Does foo

    Parameters
    ---------
    x : DataFrame
    y : DataFrame
"""
```

Of course, this would be for a different PR. The string variant is the method used to document the signature of Cython functions.

Reply:
> A decorator would be a good solution, however some of our decorators 'lose' the signature in the docs e.g. https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_stata.html

@simonjayhawkins I tried adding this one to `read_csv`and compiling. Seems ok but I can't say how robust that is